Com 1193

tresorluz opened the pull request.

ningosi committed:
Add the indicator 3 disaggreageted by entry point to be viewed in HTML view

tresorluz committed:
Added changes for the indicator3

tresorluz committed:
Added new changes for  Indicator3

tresorluz committed:
Added unit test of Indocator3 disaggregated by entry points

tresorluz committed:
COM-1193 Added unit test for CDV & TS Indicator 3

tresorluz committed:
Fixed unit test issues for CDV & TS Indicator 3 disaggregated by entrypoints

tresorluz committed:
COM-1193 merged dev into curent branch

tresorluz committed:
COM-1193 changed hiv test date concept

tmvumbi2 commented on report-testing-framework/report-testing/src/test/java/org/jembi/bahmni/report_testing/cdv_ts_reports/HivtmiIndicator3ByEntrypointsTests.java:
```diff
@@ -0,0 +1,214 @@
+                new LocalDateTime(2022, 11, 11, 8, 0),
+                ConceptEnum.POSITIVE,
+                encounterId);
+
```
@tresorluz The entry point needs to be set for this unit test

tmvumbi2 commented:
@tresorluz The indicator is not displayed in the final excel; we need to update the template to add references to display this indicator. Please remember to first merge the latest version of the dev branch in this branch, before changing the template, otherwise, there will be a merge conflict on the template that won't be resolvable (because Excel files are opaque binaries)

tmvumbi2 commented on openmrs/apps/reports/reports.json:
```diff
@@ -1956,6 +1956,12 @@
                         "sqlPath": "/var/www/bahmni_config/openmrs/apps/reports/HIV_TESTING_MONTHLY_INDICATOR_REPORT/sql/indicator1_CDV_TS_number_of_people_tested_in_the_month_disaggregation_entry_point.sql"
                     }
+                },{
+                    "name": "Report3a",
```
@tresorluz Can we use "Indicator3B" here ? The naming that I am suggesting is "Indicator", followed by its number, and finally A - for age disaggregation, and B - for entry point disaggregation

tmvumbi2 commented:
@tresorluz there is currently a merge conflict on backup-openmrs.sql, to resolve this, please merge the latest dev branch in this branch on your local PC, and locally resolve the conflict

tresorluz committed:
COM-1193 Added new sql dump

tresorluz committed:
COM-1193 Added entrypoints in the unit test

tresorluz committed:
COM-1193 Added indicator3B to the templates

tresorluz committed:
...

tresorluz committed:
COM-1192 changed the report name to Indicator3B in reports.json

tmvumbi2 committed:
Merge branch 'dev' into COM-1193

# Conflicts:
#	metadata/reportssql/hivtmi_report_functions.sql
#	report-testing-framework/data/backup-openmrs.sql
#	report-testing-framework/report-testing/src/test/java/org/jembi/bahmni/report_testing/test_utils/models/ConceptEnum.java

tmvumbi2 committed:
COM-1193 Fix template and update test db dump
Adds references for 2 entry points to the template that were missing; and updates the db dump to include the custom function for indicator 3 B
